rewrite-maven/src/main/java/org/openrewrite/maven/search/FindManagedDependency.java

```diff
@@ -99,23 +101,25 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
             @Override
             public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
                 if (isManagedDependencyTag(groupId, artifactId) &&
-                    versionIsValid(version, tag.getChildValue("version").orElse(null), versionPattern)) {
+                    versionIsValid(version, versionPattern, () -> findManagedDependency(tag, null))) {
                     return SearchResult.found(tag);
                 }
                 return super.visitTag(tag, ctx);
             }
         };
     }
 
-    private static boolean versionIsValid(@Nullable String desiredVersion, @Nullable String actualVersion,
-                                          @Nullable String versionPattern) {
+    private static boolean versionIsValid(@Nullable String desiredVersion, @Nullable String versionPattern,
+                                          Supplier<@Nullable ResolvedManagedDependency> resolvedDependencySupplier) {
         if (desiredVersion == null) {
             return true;
         }
-        if (actualVersion == null) {
-            // rare but technically valid for a dependencyManagement entry to have no version; bail if so
+        ResolvedManagedDependency resolvedDependency = resolvedDependencySupplier.get();
+        if (resolvedDependency == null) {
+            // shouldn't happen, but if it does, fail the condition
             return false;
         }
+        String actualVersion = resolvedDependency.getVersion();
```

Comment:
There's a slight difference here compared to `FindDependency` in that `actualVersion` here is nullable, as `resolvedDependency.getVersion()` can return null; Would be good to handle that properly, as the next line might otherwise throw an NPE.